mlflow/pyfunc/__init__.py

```diff
@@ -2429,12 +2429,16 @@ def predict(model_input: List[str]) -> List[str]:
                 CHAT_MODEL_INPUT_SCHEMA,
                 CHAT_MODEL_OUTPUT_SCHEMA,
             )
-            input_example = CHAT_MODEL_INPUT_EXAMPLE
+            input_example = input_example or CHAT_MODEL_INPUT_EXAMPLE
 
-            # perform output validation and throw if
-            # output is not coercable to ChatResponse
-            messages = [ChatMessage(**m) for m in input_example["messages"]]
-            params = ChatParams(**{k: v for k, v in input_example.items() if k != "messages"})
+            if isinstance(input_example, list) and isinstance(input_example[0], ChatMessage):
```

Comment:
@sunishsheth2009 If we do saves a list of messages as input example, it has different format than model signature (`CHAT_MODEL_INPUT_SCHEMA`). I wonder if that works on model endpoint, for example, the input example rendered in model endpoint UI will not be a valid input. Could you verify it that works?